pkg/cvo/upgradeable.go

```diff
@@ -33,13 +33,27 @@ const (
 
 var adminAckGateRegexp = regexp.MustCompile(adminAckGateFmt)
 
-// syncUpgradeable. The status is only checked if it has been more than
-// the minimumUpdateCheckInterval since the last check.
-func (optr *Operator) syncUpgradeable() error {
-	// updates are only checked at most once per minimumUpdateCheckInterval or if the generation changes
+// syncUpgradeable synchronizes the upgradeable status only if it has been more than
+// the minimumUpdateCheckInterval since the last synchronization or the precondition checks
+// on the payload are failing for less than minimumUpdateCheckInterval.
+func (optr *Operator) syncUpgradeable(config *configv1.ClusterVersion) error {
 	u := optr.getUpgradeable()
-	if u != nil && u.RecentlyChanged(optr.minimumUpdateCheckInterval) {
-		klog.V(2).Infof("Upgradeable conditions were recently checked, will try later.")
+	if u == nil {
+		klog.V(2).Info("Synchronizing the upgradeable status, because it's the first attempt.")
+	} else if !u.RecentlyChanged(optr.minimumUpdateCheckInterval) {
+		klog.V(2).Infof("Synchronizing the upgradeable status again, because more than %s has elapsed since %s", optr.minimumUpdateCheckInterval, u.At)
+	} else if cond := resourcemerge.FindOperatorStatusCondition(config.Status.Conditions, DesiredReleaseAccepted); cond != nil &&
+		cond.Reason == "PreconditionChecks" &&
+		cond.Status == configv1.ConditionFalse &&
+		!hasPassedDurationSinceTime(cond.LastTransitionTime.Time, optr.minimumUpdateCheckInterval) {
```

Comment:
What is the downside of just checking ` else if u != nil && !hasPassedDurationSinceTime(cond.LastTransitionTime.Time, optr.minimumUpdateCheckInterval)`

Reply:
If we leave the `cond.Reason == "PreconditionChecks"` we could remove the `cond.Status == configv1.ConditionFalse`, because the `cond.Reason == "PreconditionChecks"` [only gets set when the preconditions checks fail](https://github.com/openshift/cluster-version-operator/blob/5ad3c1434f781c68b8fab13b760d4c0c894fd21a/pkg/cvo/sync_worker.go#L349-L360), thus if the reason is `PreconditionChecks` then the status is implicitly `configv1.ConditionFalse`.

Reply:
> Since during the step PreconditionChecks the Etcd Backup is run. And while the Etcd backup is running, the ReleaseAccepted is False

> Thus the cond.LastTransitionTime.Time is probably changed after every upgrade.

Is not 100% correct. The EtcdBackup precondition check fails only in given instances:
https://github.com/openshift/cluster-version-operator/blob/c77c3360b86d8788e94493406a0d23a144e14e43/pkg/payload/precondition/clusterversion/etcdbackup.go#L76-L86

Reply:
Ok, so I think the code as it is good.  I believe we want to keep `cond.Status == configv1.ConditionFalse` in the condition. I will suggest that we change 
```
if u == nil {
		klog.V(2).Info("Synchronizing the upgradeable status because is it not set")
```